README.md

```diff
@@ -262,30 +262,126 @@ Parser consists of the following phases:
+
+## Testing
+We believe that unit tests remain the best documentation. All React components, processing logic (helpers) and actions (such as a button click) are validated with unit tests. We use Jest and Enzyme frameworks to validate both components (React) with combination with mocked storage (Redux).
+
+
+All JS files (components & helpers) have their own tests that are placed next to the tested sources. We follow the convention:
+* ```*.mock.js``` - it is configuration containing mocks for our tests
+* ```*.spec.jsx``` - it contains unit tests
+
+
+Additionally, we placed tests coverage verification in our CI. We use the jest-coverage tool for that. We decided to keep the coverage level at truly high levels (80 - 100%). It should enable future refactoring and code changes.
+
 
-### Testing
+When tests are executed, then we generate the report (test-report.xml) file in the build/test folder. Moreover, there is the coverage directory that contains the index.html file with unit tests coverage report.
+
+
+### How to run tests?
+1. run command to fire all tests: `yarn run test`
+2. run command to fire the specific test: `yarn run test [path_to_test]`
 
 ## Implementation details
 
-### Parsing HTML
+### Data flow
+An extension's architecture includes such components as:
+
+background script that contains listeners for browser events and communication with Redux
+content script that contains JavaScript that executes in the context of a page that has been loaded in the browser.
+Depending on the format (JSON or HTML markup), fragments' debug data is processed by the content script first and then stored in Redux with the background script (which basically wraps Redux store). Content script transformations are described [here](https://github.com/Knotx/knotx-fragments-chrome-extension#debug-fragment-data).
+
+```
+KNOT.x -> HTML MARKUP -> CONTENT SCRIPT -> BACKGROUND SCRIPT -> REDUX -> COMPONENTS
+                               ^
+                               |
+                         Parsing helper
+```
 
 ### Components
+The components structure in the main concept look like this:
+
+```
+•
+└── App:
+    ├── SidePanel
+    │   ├── FragmenList
+    │   │   └── FragmentListItem
+    │   │       └── NodeList
+    │   └── FragmentGannt
+    │
+    └── MainPanel
+        └── Graph
+            ├──  Timeline
+            ├──  Legend
+            │    └── LegendSection
+            └──  NodeInfo
+```
+
+#### Graph && timelines
+To create a data visualization we use vis.js library. Currently we have 3 charts in our application.
+
+* Timeline for fragments on sidePanel (FragmentGannt component)
+* Main graph for the fragment (Graph component)
+* Performance timeline for the fragment (Timeline component)
 
-#### Graph
+### Styling
+We don't use any grid system to make our app beautiful. Everything is flex. To show and hide elements we try to use a react state, without saving this information in the redux store.
```

Comment:
It is a bit not consistent: we write that we try to use a react state, in storage section we describe that we have details about expanded panels.

Reply:
I added info about sidePanel exception. This is only component witch has an expanded info stored in redux.